crates/rbuilder/src/live_builder/simulation/mod.rs

```diff
@@ -29,275 +25,46 @@ pub struct SlotOrderSimResults {
-        }
-    }
-    fn total(&self) -> usize {
-        self.mempool_txs + self.bundles + self.share_bundles
-    }
-}
-
-/// struct that continuously simulates orders.
-/// The flow is:
-/// 1 new orders are polled from new_order_sub and inserted en the SimTree
-/// 2 SimTree is polled for nonce-ready orders and are sent to be simulated (sent to  sim_req_sender)
-/// 3 simulation results are polled from sim_results_receiver and sent to slot_sim_results_sender
-/// Cancellation flow: we add every order we start to process to in_flight_orders.
-/// If we get a cancellation and the order is not in in_flight_orders we forward the cancellation.
-/// If we get a cancellation and the order is in in_flight_orders we just remove it from in_flight_orders.
-/// Only SimulatedOrders still in in_flight_orders are delivered.
-/// @Pending: implement cancellations in the SimTree.
-struct SimulationJob<DB> {
-    block_cancellation: CancellationToken,
-    /// Input orders to be simulated
-    new_order_sub: mpsc::UnboundedReceiver<OrderPoolCommand>,
-    /// Here we send requests to the simulator pool
-    sim_req_sender: flume::Sender<SimulationRequest>,
-    /// Here we receive the results we asked to sim_req_sender
-    sim_results_receiver: mpsc::Receiver<SimulatedResult>,
-    /// Output of the simulations
-    slot_sim_results_sender: mpsc::Sender<SimulatedOrderCommand>,
-    sim_tree: SimTree<DB>,
-
-    pub orders_received: OrderCounter,
-    pub orders_simulated_ok: OrderCounter,
-
-    /// Orders we got via new_order_sub and are still being processed (they could be inside the SimTree or in the sim queue)
-    /// and were not cancelled.
-    in_flight_orders: HashSet<OrderId>,
+#[derive(Debug)]
+pub struct OrderSimulationPool<DB> {
+    provider_factory: ProviderFactoryReopener<DB>,
+    running_tasks: Arc<Mutex<Vec<JoinHandle<()>>>>,
+    current_contexts: Arc<Mutex<CurrentSimulationContexts>>,
+    worker_threads: Vec<std::thread::JoinHandle<()>>,
 }
 
+/// Result of a simulation.
 #[derive(Clone, Debug)]
 pub enum SimulatedOrderCommand {
-    /// New or update order
+    /// New simulation.
     Simulation(SimulatedOrder),
+    /// Forwarded cancellation from the order source.
     Cancellation(OrderId),
 }
 
-/// Runner of the simulations.
-/// Create and call run()
-/// 1- Gets the orders orders/cancellations from new_order_sub
-/// 2- Using a SisTree checks when the orders re ready for simulation (nonce problems) and simulates them
-/// 3- Send simulation results via sim_results_receiver
-/// Also handles ShareBundle version replacements or cancellations (maybe this logic could be moved to an external object?):
-/// - Always send simulations with increasing `sequence_number`
-/// - When replacing an order we always send a cancellation for the previous one.
-/// - When we gat a cancellation we propagate it and never again send an update.
-impl<DB: Database + Clone + Send + 'static> SimulationJob<DB> {
```

Comment:
why is all of this not needed anymore?

Reply:
This is an old comment now we handle this on OrderReplacementManager.